src/plonk/circuit.rs

```diff
@@ -157,7 +160,7 @@ pub(crate) struct PointIndex(pub usize);
 /// This is a description of the circuit environment, such as the gate, wire and
 /// permutation arrangements.
 #[derive(Debug, Clone)]
-pub struct ConstraintSystem<F> {
+pub struct ConstraintSystem<C: CurveAffine, F> {
```

Comment:
These two types will never be set independently, so make this:
```suggestion
pub struct ConstraintSystem<C: CurveAffine> {
```
and then change `F` to `C::Scalar` as appropriate.

Reply:
i've introduced a `LookupData` struct so that `ConstraintSystem` no longer has to care about the curve.